### PR TITLE
Refactor attribute accessors

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This PR aims for a complete refactoring step in the abstract
shell surface API code, aiming for a big removal of code there
in favour of some abstraction.

PR #625 prepares for the next few steps I want to do here,
to refactor the attribute writing into one API call here.

Blocked by #625.

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Updated as suggested.

Shall I remove the appropriate setters/getters from the codebase in this PR as well?

**Comment by matthiasbeyer** [comment]
Targets #627 now.

**Comment by matthiasbeyer** [comment]
Branch name is inappropriate now, of course... but I guess you don't mind, do you?

**Comment by neithernut** [comment]
Nope, I don't mind.

File: src/command/object.c [hunk 1]
```diff
@@ -153,6 +153,61 @@ ws_builtin_cmd_has_attr(
...
+) {
+    if ((args[0].value.type != WS_VALUE_TYPE_OBJECT_ID) ||
+        (args[1].value.type != WS_VALUE_TYPE_STRING)) {
+        return -EINVAL;
+    }
+
+    struct ws_object* obj = ws_value_object_id_get(&args[0].object_id);
+
+    struct ws_string* name = ws_value_string_get(&args[1].string);
+    char* ident = ws_string_raw(name); // copies!
```
Comment (neithernut) [inline src/command/object.c]: You do not check whether `ident` is `NULL`.

File: src/command/object.c [hunk 2]
```diff
@@ -153,6 +153,61 @@ ws_builtin_cmd_has_attr(
...
+
+    int res = ws_object_attr_read(obj, ident, &args->value);
+    if (res != 0) {
+        goto out;
+    }
```
Comment (neithernut) [inline src/command/object.c:170]: If `res != 0`, `ident` is not `free()`d. You may fix this by simply moving the `free()`-call _above_ the if.

File: src/command/object.c [hunk 3]
```diff
@@ -153,6 +153,61 @@ ws_builtin_cmd_has_attr(
...
+
+    struct ws_object* obj = ws_value_object_id_get(&args[0].object_id);
+
+    struct ws_string* name = ws_value_string_get(&args[1].string);
+    char* ident = ws_string_raw(name); // copies!
+
+    int res = ws_object_attr_write(obj, ident, &args[2].value);
+    if (res != 0) {
+        goto out;
+    }
```
Comment (neithernut) [inline src/command/object.c:202]: Ame as above, for both, missing `NULL`-check and missing `free()`-call in case of `res != 0`.

**Comment by neithernut** [comment]
ACK for squashing.

**Comment by matthiasbeyer** [comment]
Done.

**Comment by neithernut** [comment]
ACK. You may now proceed portign the attributes of various objects to the new mechanism.

**Pull request merged**